opal/datatype/opal_convertor.c

```diff
@@ -578,8 +579,9 @@ int32_t opal_convertor_prepare_for_recv( opal_convertor_t* convertor,
     assert(! (convertor->flags & CONVERTOR_SEND));
     OPAL_CONVERTOR_PREPARE( convertor, datatype, count, pUserBuf );
 
-    if( convertor->flags & CONVERTOR_WITH_CHECKSUM ) {
-        if( !(convertor->flags & CONVERTOR_HOMOGENEOUS) ) {
+#if defined(CHECKSUM)
```

Comment:
why didn't you do the same in opal_convertor_prepare_for_send()?

Reply:
Thanks for noticing. I applied the same logic to the opal_convertor_prepare_for_send.